src/error.jl

```diff
@@ -4,6 +4,36 @@
 #
 ###############################################################################
 
+mutable struct ImpossibleInverse{T, S} <: Exception
+  data::T   # element that could not be inverted
+  mod::S    # ring or modulus with respect to which it could not be inverted
+end
+
+function ImpossibleInverse(x::T, y::S) where {T <: RingElement, S}
```

Comment:
How about "NotInvertible" instead? Also, don't most exception type end with `Error` (and a few with `Exception`)? So perhaps `NotInvertibleError`? This is also fits with saying `"x is not invertible in S" resp. "x is not invertible modulo S"

Reply:
Sure, whatever name fits with the system. Do you like the idea of throwing the separate julia `DivideError` upon division by zero? Maybe its better to have a separate error for the ultimate no-no of dividing by zero, but I don't like that it prints as "Integer division error".

Reply:
Yeah, I don't like that message either. I'd be happy with not throwing `DivideError` here, too; but I don't have a strong opinion either way.

Reply:
I am also happy with not throwing a `DivideError` in tthis case.